reana_db/alembic/versions/20231123_1342_126601b69c78_improve_indexes_usage.py

```diff
@@ -0,0 +1,71 @@
+"""Improve indexes usage.
+
+Revision ID: 126601b69c78
+Revises: b85c3e601de4
+Create Date: 2023-11-23 13:42:47.554976
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = "126601b69c78"
+down_revision = "b85c3e601de4"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    """Upgrade to 126601b69c78."""
+    # Drop old unique constraint for __reana.workflow and
+    # create new one with better column order
+    op.drop_constraint(
+        "_user_workflow_run_uc", "workflow", schema="__reana", type_="unique"
+    )
+    op.create_unique_constraint(
+        "_user_workflow_run_uc",
+        "workflow",
+        ["owner_id", "name", "run_number_major", "run_number_minor"],
+        schema="__reana",
+    )
+
+    # Create new index on (workflow_uuid, created) of __reana.job
+    op.create_index(
+        "_workflow_uuid_created_ix",
+        "job",
+        ["workflow_uuid", "created"],
+        unique=False,
+        schema="__reana",
+    )
+
+    # Create new index on (status) of __reana.workflow
+    op.create_index(
+        op.f("ix___reana_workflow_status"),
```

Comment:
Minor: is there a naming convention to be consistent with the names of the index? See `_workflow_uuid_created_ix` and `ix___reana_workflow_status`, where `ix` is once used as a prefix, and once as a suffix. I see the one for the status is autogenerated by SQLAlchemy, maybe we can try to use the same?

Reply:
Yes, SQLAlchemy autogenerates the name, but by default only for indexes. We could set up naming conventions also for foreign keys/etc. like this: https://docs.sqlalchemy.org/en/20/core/constraints.html#configuring-constraint-naming-conventions

However, we already have some inconsistencies:
```
Indexes:
    "workflow_pkey" PRIMARY KEY, btree (id_)
    "_user_workflow_run_uc" UNIQUE CONSTRAINT, btree (owner_id, name, run_number_major, run_number_minor)
    "ix___reana_workflow_status" btree (status)
Foreign-key constraints:
    "workflow_owner_id_fkey" FOREIGN KEY (owner_id) REFERENCES __reana.user_(id_)
Referenced by:
    TABLE "__reana.workflow_resource" CONSTRAINT "workflow_resource_workflow_id_fkey" FOREIGN KEY (workflow_id) REFERENCES __reana.workflow(id_)
    TABLE "__reana.workflow_session" CONSTRAINT "workflow_session_workflow_id_fkey" FOREIGN KEY (workflow_id) REFERENCES __reana.workflow(id_)
    TABLE "__reana.workspace_retention_rule" CONSTRAINT "workspace_retention_rule_workflow_id_fkey" FOREIGN KEY (workflow_id) REFERENCES __reana.workflow(id_)
```

- `workflow_pkey` and `workflow_owner_id_fkey` (generated by PostgreSQL)
- `_user_workflow_run_uc` (custom name that does not refer to any particular column)
- `ix___reana_workflow_status` (autogenerated by SQLAlchemy)

I can try to set some default naming conventions and see what happens, but I will need to update/create a migration to change the name of the already-existing constraints

Reply:
I have added a new migration to make all the names consistent between each other. Many changes, so lots of testing is needed ;)